File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        #hash tables
        counterN = Counter(nums)
        for value in counterN.values():
            if value > 1:
                return True
        return False
        # #sorting
        # nums.sort()
        # for idx in range(1,len(nums)):
        #     if nums[idx-1] == nums[idx]:
        #         return True
        # return False

File: 0217-contains-duplicate/0217-contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        //using hashset approach
        Set<Integer> set = new HashSet<>();
        boolean isDup = false;
        for(int num: nums){
            if(set.contains(num)){
                isDup = true;
                break;
            }else{
                set.add(num);
            }
        }
        return isDup;
    }
}